staNMF/staNMF.py

```diff
@@ -371,10 +392,16 @@ def instability(self, k1=0, k2=0):
                 self.instabilitydict[k] = (np.sum(distMat) / (numReplicates *
                                            (numReplicates-1)))
 
-                outputfile = open("instability.csv", "w")
-                for i in sorted(self.instabilitydict):
-                    outputfile.write(str(i) + "," +
-                                     str(self.instabilitydict[i]) + "\n")
+                if self.parallel:
```

Comment:
> I thought it might be less energetically expensive to just write this single line as a formatted string, so I don't have to initialize a csvwriter().

My experience is that it's easy to botch manual formatting. The types of errors you get with csv.writer are easy to notice and fix. Not the sneaky deadly types that manual formatting creates.